Coinjoin: add Coinjoin Settings (#1295)
zkSNACKs/WasabiDoc: docs/using-wasabi/CoinJoin.md

## Coinjoin Settings

Wasabi Wallet has automatic coinjoin.
It is done by the built-in automatic coinjoin "robot".
The robot contains instructions/settings, like how often and when to coinjoin.
The user is able to specify these settings in the `Coinjoin Settings` dialog, according to his own preferences.
The wallet ships with the default settings, which is good enough for most users.

![Wasabi Wallet Coinjoin Settings](/CoinjoinSettings.png "Wasabi Wallet Coinjoin Settings")

:::tip Coinjoin settings are wallet specific
As the coinjoin settings apply per individual wallet, it is possible to have multiple wallets with different coinjoin settings.
:::

### Automatically start coinjoin

This setting is active by default.

When this is active, the wallet will automatically start coinjoining soon after the wallet is loaded.
The wallet will coinjoin until the `privacy progress` is 100%.

When this is not active, the user will have to manually press the Play button in order to start coinjoining.

![Coinjoin Settings Automatically Start Coinjoin](/CoinjoinSettingsAutomaticallyStartCoinjoin.png "Coinjoin Settings Automatically Start Coinjoin")

### Auto-start coinjoin threshold

> The wallet will not automatically start coinjoining if the non-private balance is below the `Auto-start coinjoin threshold`, even if the `Automatically start coinjoin` is active.

Reviewer: ```suggestion
The default Auto-start coinjoin threshold is 0.01 BTC.

The wallet will not automatically start coinjoining if the non-private balance is below the `Auto-start coinjoin threshold`, even if the `Automatically start coinjoin` is active.
```

> The wallet will not automatically start coinjoining if the non-private balance is less than the `Auto-start coinjoin threshold`, even if the `Automatically start coinjoin` is active.

Reviewer: ```suggestion
The wallet will not automatically start coinjoining if the non-private balance is below the `Auto-start coinjoin threshold`, even if the `Automatically start coinjoin` is active.
```

In this case the user has to manually press Play to start coinjoining.
This setting can be used to prevent paying (relatively) high fees for smaller bitcoin amounts.

For example, if the non-private balance is 0.005 BTC and the Auto-start coinjoin threshold is 0.01 BTC, the user will have to manually press Play to start coinjoining. 

> 
> The default Auto-start coinjoin threshold is 0.01 BTC.

Reviewer: ```suggestion
```

![Auto-start Coinjoin Threshold](/AutoStartCoinjoinThreshold.png "Auto-start Coinjoin Threshold")

:::tip It is OK to set the Auto-start coinjoin threshold to 0
If you have some small non-private left overs: you can coinjoin these by pressing Play, or wait until you receive more funds.
If you want to always automatically coinjoin ALL your coins, this can be set to 0 BTC.
Note that you might pay relatively more fees for coinjoining smaller amounts.
:::

### Coinjoin Strategy

Wasabi ships with 3 coinjoin strategies: `Minimize Costs`, `Maximize Speed`, and `Maximize Privacy`.
Each of these contain different configurations.
They determine the `Anonymity score target`, `Coinjoin time preference`, and if `Red coin isolation` is enabled or not.

![Coinjoin Profile Settings](/CoinjoinProfileSettings.png "Coinjoin Profile Settings")

The default strategy is `Maximize Speed`.

> It is possible to make your own custom profile by using the customize button. 

Reviewer: ```suggestion
It is possible to make your own custom strategy by using the customize button. 
```

:::

### Automatically start coinjoin

When this is active, the wallet will automatically start coinjoining soon after the wallet is loaded.
The wallet will coinjoin until the `privacy progress` is 100%.

This setting is active by default.

When this is not active, the user will have to manually press the Play button in order to start coinjoining.

![Coinjoin Settings Automatically Start Coinjoin](/CoinjoinSettingsAutomaticallyStartCoinjoin.png "Coinjoin Settings Automatically Start Coinjoin")

### Auto-start coinjoin threshold

The wallet will not automatically start coinjoining if the non-private balance is less than the `Auto-start coinjoin threshold`, even if the `Automatically start coinjoin` is active.
If the non-private balance is below the auto-start threshold, the user has to manually press Play to start coinjoining.
This setting can be used to prevent paying (relatively high) fees for smaller bitcoin amounts.

For example, if the non-private balance is 0.005 BTC and the Auto-start coinjoin threshold is 0.01 BTC, the user will have to manually press Play to start coinjoining. 

> The default Auto-start coinjoin threshold is 0.01 BTC.

Author: I think it would be good to have all the `active by default` setting as a tip at the bottom of each setting
I can do that in follow up PR

> The default Auto-start coinjoin threshold is 0.01 BTC.

Reviewer: I moved it to the top for now, but feel free to make another PR.

![Auto-start Coinjoin Threshold](/AutoStartCoinjoinThreshold.png "Auto-start Coinjoin Threshold")

:::tip It is OK to set the Auto-start coinjoin threshold to 0
If you have some small non-private left overs: you can coinjoin these by pressing Play, or wait until you receive more funds.
If you want to always automatically coinjoin ALL your coins, this can be set to 0 BTC.
Only note is that you might pay relatively more fees for coinjoining smaller amounts.
:::

### Coinjoin Strategy

> Wasabi ships with 3 different coinjoin strategies: `Minimize Costs`, `Maximize Speed`, and `Maximize Privacy`.
> Each of these coinjoin strategies contain different configurations.

Reviewer: ```suggestion
Wasabi ships with 3 coinjoin strategies: `Minimize Costs`, `Maximize Speed`, and `Maximize Privacy`.
Each of these contain different configurations.
```

The coordinator sends this transaction over the Tor network to random Bitcoin P2P nodes, and from there it is gossiped to other nodes and miners.
Wasabi saves on mining fees by setting a confirmation target of roughly 12 hours, but you can re-register unconfirmed anonset outputs for the next round of CoinJoin.

## Coinjoin Settings

Wasabi Wallet has automatic coinjoin.
This automatic coinjoin is done by the built-in automatic coinjoin "robot".
The robot contains instructions/settings, like how much and when to coinjoin.
The user is able to specify these settings in the `Coinjoin Settings` dialog, according to his own preferences.
The wallet ships with the default settings, which is good enough for most users.

![Wasabi Wallet Coinjoin Settings](/CoinjoinSettings.png "Wasabi Wallet Coinjoin Settings")

:::tip Coinjoin settings are wallet specific

> As the coinjoin settings apply per individual wallet, it is possible to have multiple wallets with different coinjoin settings 

Reviewer 2: ```suggestion
As the coinjoin settings apply per individual wallet, it is possible to have multiple wallets with different coinjoin settings.
```

:::

### Automatically start coinjoin

> When this is active, the wallet will automatically start coinjoining when the wallet is opened.

Reviewer 2: ```suggestion
When this is active, the wallet will automatically start coinjoining soon after the wallet is loaded.
```

If the non-private balance is below the auto-start threshold, the user has to manually press Play to start coinjoining.
This setting can be used to prevent paying (relatively high) fees for smaller bitcoin amounts.

For example, if the non-private balance is 0.005 BTC and the Auto-start coinjoin threshold is 0.01 BTC, the user will have to manually press Play to start coinjoining. 

The default Auto-start coinjoin threshold is 0.01 BTC.

![Auto-start Coinjoin Threshold](/AutoStartCoinjoinThreshold.png "Auto-start Coinjoin Threshold")

:::tip It is OK to set the Auto-start coinjoin threshold to 0
If you have some small non-private left overs: you can coinjoin these by pressing Play, or wait until you receive more funds.
If you want to always automatically coinjoin ALL your coins, this can be set to 0 BTC.
Only note is that you might pay relatively more fees for coinjoining smaller amounts.
:::

### Coinjoin Strategy

Wasabi ships with 3 different coinjoin strategies: `Minimize Costs`, `Maximize Speed`, and `Maximize Privacy`.
Each of these coinjoin strategies contain different configurations.

> These coinjoin strategies determine settings like how much and when to coinjoin.

Reviewer 2: Maybe say that it specifically change anonscore target and its vlaues.

Also mention its the fee rate time preference

Author: [2d04a14](https://github.com/zkSNACKs/WasabiDoc/pull/1295/commits/2d04a14af0a3527f402e996b6bd17cc2df6b85c8)
